src/core/task-pool.cpp

```
@@ -37,76 +46,48 @@ ITaskPool::TaskHandle BlockingTaskPool::submitTask(
 
-    // Create task just to defer the payload deletion.
+    // Create a completion token for the caller.
     Task* task = new Task();
-    task->payload = payload;
-    task->payloadDeleter = payloadDeleter;
+    task->owner = this;
 
-    // Execute the task function.
     func(payload);
+    if (payloadDeleter)
+        payloadDeleter(payload);
```

Cppcheck (2.21.0)

[style] The function 'contains' is never used.

(unusedFunction)